src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs

```diff
@@ -124,14 +126,34 @@ public static async Task ExecuteAsync(GeneratedCodeWorkspace project, CodeModel
 
                 var resourceModel = new ResourceModel();
                 resourceModel.Operations = new List<ResourceOperation>();
-                resource.AllOperations.Concat(resource.ResourceCollection?.AllOperations ?? Enumerable.Empty<MgmtClientOperation>()).SelectMany(o => o).ToList().ForEach(r =>
+                var subscriptionExtension = MgmtContext.Library.GetExtension(typeof(SubscriptionResource)).AllOperations.Where(o => o.Resource?.ResourceName == resource.ResourceName);
+                var resourceGroupExtension = MgmtContext.Library.GetExtension(typeof(ResourceGroupResource)).AllOperations.Where(o => o.Resource?.ResourceName == resource.ResourceName);
+                var tenantExtension = MgmtContext.Library.GetExtension(typeof(TenantResource)).AllOperations.Where(o => o.Resource?.ResourceName == resource.ResourceName);
+                var managementGroupExtension = MgmtContext.Library.GetExtension(typeof(ManagementGroupResource)).AllOperations.Where(o => o.Resource?.ResourceName == resource.ResourceName);
```

Comment:
Why we need to get operations from these resources? Do you encounter any problem with specific RP?